client/my-sites/checkout/cart/cart-plan-ad.jsx

```diff
@@ -25,7 +25,9 @@ import { PLAN_PREMIUM } from 'lib/plans/constants';
 class CartPlanAd extends Component {
 	addToCartAndRedirect = event => {
 		event.preventDefault();
-		addItem( cartItems.premiumPlan( PLAN_PREMIUM, { isFreeTrial: false } ) );
+		const domainRegistrations = cartItems.getDomainRegistrations( this.props.cart );
```

Comment:
Need to extend this for all domain products.

Reply:
Will leave it as-is, since the upsell is only shown for registrations anyway.
Submitted an issue to track it: https://github.com/Automattic/wp-calypso/issues/23849